### ISSUE TITLE
remote component inline element-plus component

### ISSUE THREAD

**Issue by lisering** [originjs/vite-plugin-federation#251]
remote expose component:
**Button.vue**
<template>
  <div>
    <el-button>el button</el-button>
  </div>
</template>
<script setup lang="ts">
import { ElButton } from "element-plus";
</script>
<style scoped></style>

**vite.config.ts**
import { fileURLToPath, URL } from "node:url";
import federation from "@originjs/vite-plugin-federation";
import ElementPlus from "unplugin-element-plus/vite";
import cssInjectedByJsPlugin from "vite-plugin-css-injected-by-js";
import { defineConfig } from "vite";
import vue from "@vitejs/plugin-vue";
export default defineConfig({
  build: {
    target: "esnext",
    minify: false,
  },
  server: {
    port: 10000,
  },
  plugins: [
    cssInjectedByJsPlugin(),
    ElementPlus(),
    vue(),
    federation({
      name: "cbhui",
      filename: "remoteEntry.js",
      // Modules to expose
      exposes: {
        "./Button": "./src/components/Button.vue",
      },
      shared: ["vue", "pinia"],
    }),
  ],
  resolve: {
    alias: {
      "@": fileURLToPath(new URL("./src", import.meta.url)),
    },
  },
});

Uncaught (in promise) TypeError: isObject is not a function
    at buildProp (__federation_expose_Button.16438993.js:81:8)
    at __federation_expose_Button.16438993.js:171:21

**Comment by flyfishzy** [comment]
I have the same problem in [vue3-advanced-demo](https://github.com/originjs/vite-plugin-federation/tree/main/packages/examples/vue3-advanced-demo). The solution is to add the `element-plus`  to the HOST project as well, even if you don't use it yet.
When I looked deeper into the problem, I found it interesting.  Vue3 doesn't export the `isObject` function, so the chunk generated by shared doesn't export `isObject` either,  that causes the probelm. Another interesting point is that `element-plus` will have no problem using vue via CDN, because using vue via CDN will define `isObject` globally.
```
const isObject = (val) => val ! == null && typeof val === 'object';
```
To summarize, I haven't found any other better solution, so for now try to make sure the host and remote dependency libraries are consistent.

----
我在[vue3-advanced-demo](https://github.com/originjs/vite-plugin-federation/tree/main/packages/examples/vue3-advanced-demo)也遇到了同样的问题。解决办法是把`element-plus`也添加到HOST项目中，即使还没有使用它。
当我深入研究这个问题时，发现了一个有趣的现象。 Vue3没有导出`isObject`函数，所以shared生成的chunk也没有导出`isObject`，从而导致了这个问题。另一个有趣的地方是，`element-plus`通过CDN使用vue不会有问题，因为通过CDN使用vue会全局定义`isObject`。
```
const isObject = (val) => val ! == null && typeof val == 'object';
```
总结一下，我还没有找到其他更好的解决方案，所以现在要尽量确保主机和远程依赖库的一致性。

**Comment by dqdeep** [comment]
在host工程中也使用 ElementUI 这样在 __federation_shared_vue.js 文件中就会导出 isObject 这个函数

**Comment by dadaguai-git** [comment]
我也遇到这种问题 当我remote模块依赖ui组件是 host端调用 样式总是丢失  除了在host端安装相同依赖还有其他办法么
